channel/channel.go

```diff
@@ -112,6 +122,23 @@ func (c *Channel) UnmarshalJSON(data []byte) error {
 	return nil
 }
 
+// getStatusFromChannel returns the status of the channel
+func (c *Channel) Status() Status {
+	if c.FinalSignedByMe() {
+		if c.FinalCompleted() {
+			return Closed
+		}
+		return Closing
+	}
+	if c.PostFundSignedByMe() {
+		if c.PostFundComplete() {
+			return Open
```

Comment:
In the case of an intermediary `c.PostFundComplete` could never be completed, so they would never get back a status of `Open`.

Reply:
I think it could be completed some times, but you are right that it would often remain incomplete.

I think it is OK -- but for sure it is leaking something which seems like an implementation detail across the API.

We have a mixture of information from the channel and from the objective. As long as we are focussed on the channel (and want to rip out any dependence on objective), I think we would probably have to be ok with that.